test/test_fip81.py

```diff
@@ -0,0 +1,104 @@
+import os
+import unittest
+from datetime import date, timedelta
+from dotenv import load_dotenv
+
+
+from jax import config
+config.update("jax_enable_x64", True)
+
+import mechafil_jax.data as data
+import mechafil_jax.supply as jax_supply
+import mechafil_jax.constants as C
+import mechafil_jax.sim as sim
+import scenario_generator.utils as u
+
+import numpy as np
+import jax.numpy as jnp
+import tqdm.auto as tqdm
+import pickle
+import json
+
+
+class TestFip81(unittest.TestCase):
+    def get_offline_data(self, start_date, current_date, end_date):
+        # Load the .env file
+        load_dotenv()
+        # Get the path to the authentication file from the environment variable
+        auth_file_path = os.getenv('SPACESCOPE_AUTH_PATH')
+        
+        if auth_file_path:
+            # Load the token from the JSON file
+            with open(auth_file_path, 'r') as f:
+                auth_data = json.load(f)
+        
+            # Assuming the key in the JSON file is "auth_key" and it contains the Bearer token
+            PUBLIC_AUTH_TOKEN = auth_data.get('auth_key')
+            
+            if PUBLIC_AUTH_TOKEN:
+                pass
+            else:
+                raise Exception("The 'auth_key' is missing in the JSON file.")
+        else:
+            raise Exception("SPACESCOPE_AUTH_PATH environment variable is not set.")
+        
+        offline_data = data.get_simulation_data(PUBLIC_AUTH_TOKEN, start_date, current_date, end_date)
+
+        _, hist_rbp = u.get_historical_daily_onboarded_power(current_date - timedelta(days=180), current_date)
+        _, hist_rr = u.get_historical_renewal_rate(current_date - timedelta(days=180), current_date)
+        _, hist_fpr = u.get_historical_filplus_rate(current_date - timedelta(days=180), current_date)
+        smoothed_last_historical_rbp = float(np.median(hist_rbp[-30:]))
+        smoothed_last_historical_rr = float(np.median(hist_rr[-30:]))
+        smoothed_last_historical_fpr = float(np.median(hist_fpr[-30:]))
+
+        result = (
+            offline_data, smoothed_last_historical_rbp, smoothed_last_historical_rr,
+            smoothed_last_historical_fpr, hist_rbp, hist_rr, hist_fpr
+        )
+        return result
+
+    def test_fip81(self):
+
+        # Load data from original fip81 implementation
+        test_dir = os.path.dirname(__file__)  # This will get the directory of the current test file
+        file_path = os.path.join(test_dir, 'results_original_branch.pkl')
```

Comment:
this is opaque, so it would be better if we can have the code which generates this pickle in the unittest file.  otherwise, it is hard to tell what the reference actually is.